### PR TITLE
Move models and exceptions to common module

### PR DESCRIPTION
**Pull request by gandreadis**
In order to be able to use them in combination (nesting) but also to prevent and reduce duplication, I'm moving them to common. I initially didn't want to move the exceptions, but since the models depend on them, I also moved them.

### CODE REVIEW

File: common/build.gradle [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apply plugin: "org.springframework.boot"
+apply plugin: 'io.spring.dependency-management'
+
+dependencies {
+    implementation 'org.springframework.boot:spring-boot-starter-web'
```
Comment (qurben) [inline common/build.gradle]: spring boot starter web requires a main class

Comment (qurben) [reply common/build.gradle]: Or spring in general

Comment (gandreadis) [reply common/build.gradle]: Hehe thanks :)

Comment (gandreadis) [reply common/build.gradle]: @qurben I've tried adding one, but it doesn't seem to resolve the issue, could you have a look?

**Comment by casperboone** [comment]
We don't need a separate subproject for this. The root is already a project and seems to be a logical place for this. So we can just create a `src/` directory. Since I only applied the gradle config to subprojects (on purpose), this also makes sure that Spring etc. won't be enabled there :)

**Comment by gandreadis** [comment]
@qurben Are you fine with me moving all commons to `/src/` (since you created `common` in one of your earlier PRs)?

**Comment by qurben** [comment]
I'm fine with that. 

Is subprojects referencing the main project in gradle common practise? The reverse (main project referencing subprojects) would be what I expect.

**Comment by qurben** [comment]
Or the way it's done in [Project Lib Dependencies](https://docs.gradle.org/current/userguide/multi_project_builds.html#sec:project_jar_dependencies), and create a directory for all services so they can be grouped easily.

**Comment by casperboone** [comment]
That is indeed a lot nicer, however, I tried to keep our setup as easy as possible, and therefore chose to put the different projects in the root. Personally, I'm fine with both though

**Comment by gandreadis** [comment]
Won't moving all of them into one directory cause even more chaos with existing PRs though?

**Comment by casperboone** [comment]
Yes, so that might be a reason to hold off on that for now. Something we can consider as a refactor later though.

**Comment by gandreadis** [comment]
How about keeping them in `common` for now then? I'm getting all sorts of Gradle errors when I try to move common to root, and I personally also think having `common` as an explicit module is clearer atm than having one src folder next to 4 modules (in terms of mixing abstractions)

**Comment by gandreadis** [comment]
If you agree, I'll need an approving review on this PR :)

**Pull request merged**